### PR TITLE
Self closing tag

### LINKED ISSUE #2: Xml.decode doens't handle self-closing tags
In XML, when a tag has no children it can be written as self-closing. Actually, it's not only is it valid; sometimes the spec calls for it.<sup>[source](https://www.w3.org/TR/REC-xml/#sec-starttags)</sup> The syntax is as follows: `<tag></tag>` is equivalent to `<tag />`

Unfortunately, `Xml.decode` doesn't handle this possibility. For instance, if you try to decode
```
<tag>
    <autoclosing />
    <autoclosing />
</tag>
```

you get

```
StrNode "\n <autoclosing />\n <autoclosing />\n "
```

when it's supposed to get

```
Object (
    [ Tag "noautoclosing" (Dict.fromList []) (Object [])
    , Tag "noautoclosing" (Dict.fromList []) (Object []) ]
)
```

<hr />

You can get an reproducible example [here](https://ellie-app.com/Y4YFzczzMHa1/0).

### PR DESCRIPTION
**Pull request by kitofr**
This should close #2 ... hopefully :)

Thanks to @mateusfccp in #4 for fixing the majority of things :)

### CODE REVIEW

File: tests/Tests.elm [hunk 1]
```diff
@@ -27,6 +26,28 @@ exampleAsString =
 """
         |> String.trim
 
+selfClosingExampleAsString : String
+selfClosingExampleAsString =
+  """
+<person>
+    <name is="me">kitofr</name>
+    <here is="false" />
+    <there is="true" />
```
Comment (eeue56) [inline tests/Tests.elm]: What about when there are two tags, self-closing, one after the other? e.g `<there/><there/>`?

Comment (kitofr) [reply tests/Tests.elm]: Think that worked as well... can try.

Comment (eeue56) [reply tests/Tests.elm]: Would be good to have a case for it too, just in case

**Comment by eeue56** [comment]
lgtm! thanks!

**Comment by eeue56** [comment]
published as 2.1.1

**Comment by mateusfccp** [comment]
Who would guess it would be a `+ 1?`. Thanks for solving the issue!

**Pull request merged**